### PR TITLE
History fixes and misc about page fixes

### PR DESCRIPTION
**Pull request by bsclifton**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

-----

redo of https://github.com/brave/browser-laptop/pull/5559

**NOTE**: _I thought I had fixed the history issues with this, but I was wrong. I removed the issues I thought I had fixed from this PR_

-----

- History code was refactored
  - one step forward with regard to https://github.com/brave/browser-laptop/issues/3502
  - session helper created for aboutHistory (tests added)
  - history specific methods broke out into historyUtil (tests added)
  - history stores it's data in AppStore.about.history (similar structure to new tab code)

- about:history is only updated when something changed
  - number of calls to render on about:history is drastically reduced (should fix https://github.com/brave/browser-laptop/issues/5382; please reopen if not)

- about:brave text update
Implements followup feedback for https://github.com/brave/browser-laptop/pull/5436#issuecomment-259144537

Auditors: @cezaraugusto, @darkdh, @bbondy 

-----

Remove unnecessary/broken/useless context menu items for about pages

Fixes https://github.com/brave/browser-laptop/issues/4812

-----

Update icon for about pages to fa-list (except for about:newtab)

Fixes https://github.com/brave/browser-laptop/issues/5497

Auditors: @bradleyrichter

Test Plan:
1. Launch Brave and visit about:about
2. Open each of the links and confirm it's the `fa-list` icon

## Screenshots
Here's one showing:
- fixing the casing on component names in about:brave
- new "fa-list" urlbar icon
- context menu has  items removed from about pages
  - save as
   - print
   - view source
   - inspect element

screen shot 2016-11-11 at 5 13 56 pm

### CODE REVIEW

**Comment by bsclifton** [comment]
Comment per @bradleyrichter in last thread:
> It looks like our icons are riding high for some reason. But this will look nice.

(I'll look at fixing that up :smile: )

File: app/common/lib/historyUtil.js [hunk 1]
```diff
@@ -2,7 +2,26 @@
...
+  if (left.get('lastAccessedTime') < right.get('lastAccessedTime')) return 1
+  if (left.get('lastAccessedTime') > right.get('lastAccessedTime')) return -1
+  return 0
+}
+
+module.exports.getHistory = (sites) => {
+  sites = makeImmutable(sites) || new Immutable.List()
+  return sites.filter((site) => siteUtil.isHistoryEntry(site))
+      .sort(sortTimeDescending)
+      .slice(0, aboutHistoryMaxEntries)
```
Comment (bsclifton) [inline app/common/lib/historyUtil.js]: fix from https://github.com/brave/browser-laptop/commit/a9c3304ec4a646ea27da1a4de7ff47082a1d9481 was kept here when rebasing :smile:

File: test/unit/lib/historyUtilTest.js [hunk 2]
```diff
@@ -35,6 +35,29 @@ describe('historyUtil', function () {
...
+    })
+    it('sorts the items by date/time DESC', function () {
+      const result = historyUtil.getHistory(historyMultipleDays)
+      const expectedResult = historyDayThree.toJS().reverse()
+      expectedResult.push(historyDayTwo.toJS())
+      expectedResult.push(historyDayOne.toJS())
+      assert.deepEqual(result.toJS(), expectedResult)
+    })
+    it('only returns `historyUtil.maxEntries` results', function () {
+      let tooManyEntries = new Immutable.List().push(historyDayOne)
```
Comment (bsclifton) [inline test/unit/lib/historyUtilTest.js]: note to self: this needs to check the order they're in

File: app/common/lib/historyUtil.js (hunk 1 above)
Comment (bbondy) [inline app/common/lib/historyUtil.js]: 👍

**Comment by bbondy** [comment]
I think it looks good. I also asked @darkdh to take a look and those can be done in followups if he has comments.

**Comment by darkdh** [comment]
lgtm, ++

**Pull request merged**